### PR TITLE
VTM Project Foundational Reboot

### PR DESCRIPTION
**Pull request by andrewf-bjss**
PR to merge in the VTM project foundational reboot from our feature branch

### CODE REVIEW

**Review by danielchildsbjss — DISMISSED** [review]
This code was all reviewed in a long running feature branch

**Comment by premghinde** [comment]
Please remove the archive folder. If you need to get back to historical code it is all saved within git history.

**Review by NiallBunting — APPROVED** [review]

**Comment by andrewf-bjss** [comment]
> Please remove the archive folder. If you need to get back to historical code it is all saved within git history.

Agreed, thanks Prem - removed  👍🏻

**Review by danielchildsbjss — APPROVED** [review]

**Review by NiallBunting — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -1,174 +1,104 @@
...
-    "microsoft-adal-angular6": "^1.3.0",
-    "ngrx-forms": "3.2.1",
-    "ngrx-store-freeze": "^0.2.4",
-    "ngx-filesaver": "^2.2.1",
-    "ngx-json-viewer": "^2.4.0",
-    "rxjs": "6.2.2",
-    "rxjs-compat": "^6.5.3",
-    "uuid": "^7.0.2",
-    "zone.js": "0.8.25"
+    "@angular/animations": "~13.2.0",
```
Comment (premghinde) [inline package.json:30]: Normalise all versions to either tilde, caret or fixed.

**Review by mcadenas-bjss — APPROVED** [review]

File: src/app/app.component.spec.ts [hunk 2]
```diff
@@ -1,39 +1,38 @@
...
-  }
-
-  return obj;
-};
+import { UserService } from './services/user-service/user-service';
+import { StoreModule } from '@ngrx/store';
+import { AppComponent } from './app.component';
+import { FooterComponent } from './layout/footer/footer.component';
+import { HeaderComponent } from './layout/header/header.component';
+import { MsalModule, MsalBroadcastService, } from "@azure/msal-angular";
```
Comment (premghinde) [inline src/app/app.component.spec.ts:8]: Linter not picking up double quotes?

File: src/app/features/home/home.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    .compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(HomeComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
```
Comment (premghinde) [inline src/app/features/home/home.component.spec.ts:25]: I feel there can be a greater number of tests for this component.

File: package.json (hunk 1 above)
Comment (andrewf-bjss) [reply package.json]: Normalised to ^

File: src/app/features/home/home.component.spec.ts (hunk 3 above)
Comment (andrewf-bjss) [reply src/app/features/home/home.component.spec.ts:25]: We're waiting on the service design before we spend too much time on the home component

File: src/app/pipes/default-null-or-empty.pipe.spec.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  it('transforms null to "-"', () => {
+    expect(pipe.transform(null)).toBe('-');
+  });
+
+  it('transforms "" to "-"', () => {
+    expect(pipe.transform('')).toBe('-');
+  });
+
+  it('transforms "     " to "-"', () => {
+    expect(pipe.transform('')).toBe('-');
```
Comment (premghinde) [inline src/app/pipes/default-null-or-empty.pipe.spec.ts:16]: test input doesn't match the spec.

File: package.json [hunk 5]
```diff
@@ -1,174 +1,104 @@
...
-    "jest-runner-tsc": "^1.6.0",
-    "jest-sonar-reporter": "^2.0.0",
-    "prettier": "1.19.1",
-    "reflect-metadata": "^0.1.13",
-    "ts-loader": "^6.2.1",
-    "ts-node": "5.0.1",
-    "tslint": "5.9.1",
-    "typescript": "2.9.2"
+    "@angular-devkit/build-angular": "^13.2.4",
+    "@angular-eslint/builder": "13.1.0",
```
Comment (premghinde) [inline package.json:52]: all normalised

File: src/app/vehicle-technical-record/vehicle-technical-record.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    .compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(VehicleTechnicalRecordComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
```
Comment (premghinde) [inline src/app/vehicle-technical-record/vehicle-technical-record.component.spec.ts:20]: Can add more tests?

File: src/environments/environment.prod.ts [hunk 7]
```diff
@@ -1,6 +1,7 @@
 export const environment = {
-  name: 'prod',
   production: true,
-  APIServerUri: 'http://localhost:3005',
-  APITestResultServerUri: 'http://localhost:3006'
+  VTM_CLIENT_ID: '',
```
Comment (premghinde) [inline src/environments/environment.prod.ts:3]: Should these values not be parameterised?

File: src/app/features/search/search.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    fixture.detectChanges();
+
+    jest.clearAllMocks();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should call the service to search by VIN', () => {
```
[the remark this answers is not in the document]
Comment (andrewf-bjss) [reply src/app/features/search/search.component.spec.ts:36]: HTTP error test case added to the [technical-record.service.spec](https://github.com/dvsa/cvs-app-vtm/blob/feature/CB2-2942/src/app/services/technical-record/technical-record.service.spec.ts)

File: src/environments/environment.prod.ts (hunk 7 above)
Comment (andrewf-bjss) [reply src/environments/environment.prod.ts:3]: Sorry @premghinde, can you elaborate please?

File: src/polyfills.ts [hunk 9]
```diff
@@ -8,64 +8,45 @@
  *      file.
  *
  * The current setup is for so-called "evergreen" browsers; the last versions of browsers that
- * automatically update themselves. This includes Safari >= 10, Chrome >= 55 (including Opera),
- * Edge >= 13 on the desktop, and iOS 10 and Chrome on mobile.
+ * automatically update themselves. This includes recent versions of Safari, Chrome (including
```
Comment (premghinde) [inline src/polyfills.ts:11]: It saddens me to say that Angular browser support may be ahead of gov.uk requirements. https://www.gov.uk/service-manual/technology/designing-for-different-browsers-and-devices#browsers-to-test-in-from-june-2021

File: tsconfig.spec.json [hunk 10]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (premghinde) [inline tsconfig.spec.json:1]: What is the purpose of this file?

File: src/environments/environment.prod.ts (hunk 7 above)
Comment (premghinde) [reply src/environments/environment.prod.ts:3]: Is the client id likely to be an empty string in production?

File: src/polyfills.ts (hunk 9 above)
Comment (andrewf-bjss) [reply src/polyfills.ts:11]: @premghinde - is this a concern for us as VTM won't be a GOV.UK externally facing service?

File: src/environments/environment.prod.ts (hunk 7 above)
Comment (andrewf-bjss) [reply src/environments/environment.prod.ts:3]: I see, thank you - no, the build pipeline substitutes this file with one containing production values, this exists in the repo for testing production builds locally

Comment (premghinde) [reply src/environments/environment.prod.ts:3]: Got it. I wasn't aware of that step.

File: tsconfig.spec.json (hunk 10 above)
Comment (andrewf-bjss) [reply tsconfig.spec.json:1]: it extends the Typescript config with unit test specific params

File: src/app/vehicle-technical-record/vehicle-technical-record.component.spec.ts (hunk 6 above)
Comment (andrewf-bjss) [reply src/app/vehicle-technical-record/vehicle-technical-record.component.spec.ts:20]: The Vehicle Tech Record page / component is just a placeholder at this stage to get us to the Test Record - the Tech Records team will be replacing, re-working or adding to this component in upcoming sprints, and, I'm sure, will add more tests as appropriate  👍🏻

File: src/polyfills.ts (hunk 9 above)
Comment (premghinde) [reply src/polyfills.ts:11]: Hurrah! I've found that all internal users have Edge as the default browser within DVSA, so it pleases me greatly to say you can indeed lose the legacy support.

Comment (andrewf-bjss) [reply src/polyfills.ts:11]: Thanks @premghinde, that's made my day!

Comment (premghinde) [reply src/polyfills.ts:11]: Believe me, the last thing I would have wanted to get you to do was reintroduce the legacy support.

File: sonar-project.properties [hunk 11]
```diff
@@ -1,20 +0,0 @@
-#----- Default SonarQube server
```
[the remark this answers is not in the document]
Comment (andrewf-bjss) [reply sonar-project.properties:1]: Done, thanks @premghinde  - #282 👍🏻

**Pull request merged**